liblumen_alloc/src/erts/term/binary/maybe_aligned_maybe_binary.rs

```diff
@@ -252,3 +254,44 @@ macro_rules! impl_maybe_aligned_try_into {
 
 impl_maybe_aligned_try_into!(MatchContext);
 impl_maybe_aligned_try_into!(SubBinary);
+
+#[derive(Debug, Error)]
+pub enum TryIntoByteVecError {
```

Comment:
Are there any other ways this conversion can fail? Might be more generally useful to make it a struct and call it `InvalidBinaryError` - we can then use it in more places, and use the ability to add contex to errors via `anyhow` to provide more contextual information if it is necessary